#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.41209565340584303 -0.05342150436974051 0.9095731500626462 3.37633840810016
  position 0.6151372360707592 0.05981260874604931 0.44459327467643595
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation -0.00563652 -0.00621327 0
}
Robot {
  children [
    DistanceSensor {
      translation 0.04 0.015 0.02
      rotation 0 0 1 0.26
      children [
        DEF dist_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ps_0"
    }
    DistanceSensor {
      translation 0.04 -0.015 0.02
      rotation 0 0 1 -0.26
      children [
        USE dist_sensor
      ]
      name "ps_1"
    }
    DEF Body Pose {
      translation 0 0 0.0415
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0.045 0.025
      }
      device [
        RotationalMotor {
          name "motor_1"
        }
        PositionSensor {
          name "wheel_enc_1"
        }
      ]
      endPoint Solid {
        translation 0 0.045 0.025
        rotation 1 0 0 1.57
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "wheel_1"
        boundingObject USE wheel
        physics DEF wheel_ph Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 -0.045 0.025
      }
      device [
        RotationalMotor {
          name "motor_2"
        }
        PositionSensor {
          name "wheel_enc_2"
        }
      ]
      endPoint Solid {
        translation 0 -0.045 0.025
        rotation 1 0 0 1.57
        children [
          USE wheel
        ]
        name "wheel_2"
        boundingObject USE wheel
        physics USE wheel_ph
      }
    }
    Solid {
      translation 0.0415 0.015 0.07
      children [
        DEF eye Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "eye_1"
    }
    Solid {
      translation 0.0415 -0.015 0.07
      children [
        USE eye
      ]
      name "eye_2"
    }
  ]
  boundingObject USE Body
  physics Physics {
  }
  controller "avoid_obstacle"
}
